# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CanvasRenderer.h
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 gfx/layers/CanvasRenderer.h
--- a/gfx/layers/CanvasRenderer.h	Wed Jan 08 21:52:03 2020 +0000
+++ b/gfx/layers/CanvasRenderer.h	Wed Jan 08 22:19:14 2020 +0000
@@ -25,6 +25,7 @@
 class AsyncCanvasRenderer;
 class ClientCanvasRenderer;
 class CopyableCanvasRenderer;
+class OOPCanvasRenderer;
 class PersistentBufferProvider;
 class WebRenderCanvasRendererAsync;
 
@@ -32,11 +33,12 @@
   CanvasInitializeData();
   ~CanvasInitializeData();
 
-  // One of these three must be specified for Canvas2D, but never more than one
+  // One of these four must be specified for Canvas2D, but never more than one
   RefPtr<PersistentBufferProvider>
       mBufferProvider;  // A BufferProvider for the Canvas contents
   RefPtr<mozilla::gl::GLContext> mGLContext;  // or this, for GL.
   RefPtr<AsyncCanvasRenderer> mRenderer;      // or this, for OffscreenCanvas
+  RefPtr<OOPCanvasRenderer> mOOPRenderer;  // or this, for WebGL in host process
 
   typedef void (*TransactionCallback)(void* closureData);
   TransactionCallback mPreTransCallback = nullptr;